pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,655 @@
+	}
+
+	return downloadUrl, nil
+}
+
+// waitForVirtualMachineExport waits for the VirtualMachineExport status and external links to be ready
+func waitForVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExportInfo, interval, timeout time.Duration) error {
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		vmexport, err := getVirtualMachineExport(client, vmeInfo)
+		if err != nil {
+			return true, err
+		}
+
+		if vmexport == nil {
+			return true, err
+		}
+
+		if vmexport.Status == nil {
+			return false, nil
+		}
+
+		if vmexport.Status.Phase != exportv1.Ready {
+			fmt.Printf("Waiting for VM Export %s status to be ready...\n", vmeInfo.Name)
+			return false, nil
+		}
+
+		if vmexport.Status.Links == nil || vmexport.Status.Links.External == nil {
+			fmt.Printf("Waiting for VM Export %s external links to be available...\n", vmeInfo.Name)
+			return false, nil
+		}
+
+		fmt.Printf("Processing completed successfully\n")
+		return true, nil
+	})
+
+	return err
+}
+
+// handleHTTPRequest generates the GET request with proper certificate handling
+func handleHTTPRequest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, downloadUrl string, insecure bool) (*http.Response, error) {
+	token, err := getTokenFromSecret(client, vmexport)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new certPool and append our external SSL certificate
+	cert := vmexport.Status.Links.External.Cert
+	roots := x509.NewCertPool()
+	roots.AppendCertsFromPEM([]byte(cert))
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: roots},
+	}
+	httpClient := httpClientCreatorFunc(transport, insecure)
+
+	// Generate and do the request
+	req, _ := http.NewRequest("GET", downloadUrl, nil)
+	req.Header.Set(exportTokenHeader, token)
+
+	return httpClient.Do(req)
+}
+
+// getHTTPClient assigns the default, non-mocked HTTP client
+func getHTTPClient(transport *http.Transport, insecure bool) *http.Client {
+	if insecure == true {
+		transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: insecure,
+			},
+		}
+	}
+
+	client := &http.Client{Transport: transport}
+	return client
+}
+
+// copyFileWithProgressBar serves as a wrapper to copy the file with a progress bar
+func copyFileWithProgressBar(output *os.File, resp *http.Response) error {
+	barTemplate := fmt.Sprintf(`{{ "Downloading file:" }} {{counters . }} {{ cycle . %s }} {{speed . }}`, progressBarCycle)
+
+	// start bar based on our template
+	bar := pb.ProgressBarTemplate(barTemplate).Start(0)
+	defer bar.Finish()
+	rd := bar.NewProxyReader(resp.Body)
+	bar.Start()
+
+	_, err := io.Copy(output, rd)
+	return err
+}
+
+// getOrCreateTokenSecret obtains a token secret to be used along with the virtualMachineExport
+func getOrCreateTokenSecret(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) (*k8sv1.Secret, error) {
+	// Securely randomize a 20-bytes token
+	token := make([]byte, secretTokenLenght)
```

Comment:
bytes are hard to work with, strings are better

Reply:
Yeah, I think I'll do the same as in https://github.com/kubevirt/kubevirt/pull/8418 and rebase the other PR once we merge this one. I'll force-push this change now.